src/content/reference/react/index.md

```diff
@@ -10,6 +10,14 @@ This section provides detailed reference documentation for working with React. F
 
 The React reference documentation is broken down into functional subsections:
 
+## Rules of React {/*rules-of-react*/}
+
+JavaScript rendering libraries and frameworks like React have constraints or "rules" to make the programming model cohesive and easy to reason about, while also helping you prevent bugs in your code. The rules also have the added benefit of creating a safe space for React to optimize and run your code more efficiently.
```

Comment:
Everything in this paragraph is true, but it also feels like it downplays or is missing the most important piece: _the rules make React what it is_. It isn't that there is using React and separately that there are some rules around it, the rules _define_ React (as a programming model).

As the text here points out, this is also true of other rendering libraries (though I think React's model is particular coherent if you approach it from a more formal perspective). 

One way to reconcile this in the text would be something like:

React is a programming paradigm for declarative user interfaces, in which developers describe what should be rendered given the current state and the framework is responsible for updating the UI. Just as different programming languages have their own ways of expressing concepts, React has its own idioms — or rules — for how to express patterns in a way that is easy to understand and yields high-quality applications.

To learn more about expressing UIs with React, we recommend reading [Thinking in React]. This section describes the React programming paradigm from the perspective of its key rules, which may differ from approaches you've used elsewhere. Many of these rules are common to other UI libraries and non-UI related "reactive" systems, in addition to being a strong approach to building robust software generally.

Reply:
Thanks, incorporated!